web/controllers/v1/trigger_execution_controller.ex

```diff
@@ -0,0 +1,166 @@
+defmodule Cog.V1.TriggerExecutionController do
+  use Cog.Web, :controller
+
+  import Cog.Plug.Util, only: [get_request_id: 1,
+                               get_user: 1,
+                               set_user: 2]
+
+  require Logger
+
+  alias Cog.Plug.Authentication
+
+  alias Cog.Models.User
+  alias Cog.Models.Trigger
+  alias Cog.Repository.Triggers
+  alias Cog.Adapters.Http.AdapterBridge
+
+  plug :parse
+
+  def execute_trigger(conn, %{"id" => trigger_id}) do
```

Comment:
This function is kinda hard for me to follow. Would it be clearer if it was broken up a bit? Maybe using `with` and some more intermediary functions?